### PR TITLE
[EN DateTimeV2] Support for [n] [dateUnit] from [date] (#1564)

### PR DESCRIPTION
**Pull request by LionbridgeCSII**
Added support for cases like "3 days before Monday", "4 weeks after January 15th", "5 months from next Friday" (#1564).
Relevant test cases added to DateTimeModel

### CODE REVIEW

**Comment by tellarin** [comment]
@LionbridgeCSII can you resolve conflicts here and see if this fix still applies? Thanks.

**Comment by LionbridgeCSII** [comment]
> @LionbridgeCSII can you resolve conflicts here and see if this fix still applies? Thanks.

Conflicts solved.
Yes, the fix still applies (the targeted patterns fail without it).

**Comment by tellarin** [comment]
Can you re-resolve the conflicts here and see if the build still passes?

File: .NET/Microsoft.Recognizers.Text.DateTime/Extractors/BaseDateExtractor.cs [hunk 1]
```diff
@@ -560,6 +560,26 @@ private List<Token> ExtractRelativeDurationDate(string text, DateObject referenc
                 if (match.Success)
                 {
                     ret.AddRange(AgoLaterUtil.ExtractorDurationWithBeforeAndAfter(text, er, ret, Config.UtilityConfiguration));
+
+                    // Check for combined patterns Duration + Date, e.g. '3 days before Monday', '4 weeks after January 15th'
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.DateTime/Extractors/BaseDateExtractor.cs:564]: Could this perhaps be moved to the new merged method or it doesn't make sense?

File: Specs/DateTime/English/DateTimeModel.json [hunk 2]
```diff
@@ -18358,5 +18358,101 @@
         }
       }
     ]
+  },
+  {
+    "Input": "It will happen 3 days from Tuesday.",
```
Comment (tellarin) [inline Specs/DateTime/English/DateTimeModel.json:18363]: Can you add some cases with "from" and also add them in Experimental mode to show there's no side-effect of the interactions with the interpretation of "from" there?

Comment (tellarin) [reply Specs/DateTime/English/DateTimeModel.json:18363]: @LionbridgeCSII

Comment (LionbridgeCSII) [reply Specs/DateTime/English/DateTimeModel.json:18359]: added.

File: .NET/Microsoft.Recognizers.Text.DateTime/Extractors/BaseDateExtractor.cs (hunk 1 above)
Comment (LionbridgeCSII) [reply .NET/Microsoft.Recognizers.Text.DateTime/Extractors/BaseDateExtractor.cs:564]: which merged method do you mean?

Comment (tellarin) [reply .NET/Microsoft.Recognizers.Text.DateTime/Extractors/BaseDateExtractor.cs:564]: MergeAllResults in PR #2305.

Comment (LionbridgeCSII) [reply .NET/Microsoft.Recognizers.Text.DateTime/Extractors/BaseDateExtractor.cs:564]: I see, but MergeAllResults is analogous to MergeAllTokens which only merges overlapping extractions. For me, this seems more appropriate in the current method ExtractRelativeDurationDate which already deals with similar patterns like "3 days from today".

**Review by tellarin — APPROVED** [review]

**Pull request merged**